### PR TITLE
docs: Adds menu mdx docs

### LINKED ISSUE #1164: Review and Update Popups: Menu Readme File to MDX
## 💡 Suggestion

Review the `README` file for the Menu (preview) component and:
- [ ] Copy/convert it to a `menu/stories/Menu.stories.mdx` file.
- [ ] Test code snippets and examples to ensure they are up-to-date and accurate, update as needed. Examples should go in a `menu/stories/examples` folder.
- [ ] Delete the component `stories.tsx` file.
- [ ] Fix any Cypress tests that no longer work due to changes in the stories.
- [ ] Update the `README` to link to Storybook and the current Design Site, and empty the rest of the content out ([example](https://github.com/Workday/canvas-kit/blob/master/modules/react/text-input/README.md)).
- [ ] Update Storybook links in the current Design Site (or ping @jamesfan and he can handle it).

Example PRs where I did similar work converting READMEs to MDXes: [Inputs](https://github.com/Workday/canvas-kit/pull/1128), [Indicators](https://github.com/Workday/canvas-kit/pull/1169), [Card and Table](https://github.com/Workday/canvas-kit/pull/1185)

Refer to [Standardize MDX Structure](https://github.com/Workday/canvas-kit/discussions/1131) on how to structure your MDX document.

Even though Menu isn't a Compound Component, its MDX should use the Compound Component structure (search the Standardize doc for "Usage: Compound Component") since it includes multiple components that are composed together (`Menu` and `MenuItem`).

### PR DESCRIPTION
**Pull request by sahlhoff**
fixes #1164

### CODE REVIEW

File: modules/preview-react/menu/stories/Menu.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+import {Icons} from './examples/Icons';
+import {ManyItems} from './examples/ManyItems';
+
+<Meta title="Preview/Menu/React" component={Menu} />
+
+# Canvas Kit Menu
+
+<a href="https://github.com/Workday/canvas-kit/tree/master/modules/preview-react/README.md">
+  <img src="https://img.shields.io/badge/PREVIEW-beta-blueviolet" alt="PREVIEW: Beta" />
+</a>
```
Comment (sahlhoff) [inline modules/preview-react/menu/stories/Menu.stories.mdx:17]: Left the badge here...LMK if I should remove it 👍

Comment (alanbsmith) [reply modules/preview-react/menu/stories/Menu.stories.mdx:17]: Good catch. There's some [discussion](https://github.com/Workday/canvas-kit/discussions/1131#discussioncomment-950500) on this. I removed it in my Breadcrumbs PR. I think probably remove it for now, and when we have a good way to display this on the docs site, we can add it back.

File: modules/preview-react/menu/stories/Menu.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+- Moves focus to the last menu item.
+
+### A-Z / a-z
+
+- Moves focus to the next menu item with a label that starts with the typed character if such an
+  menu item exists.
+- Otherwise, focus does not move.
+
+## Menu Component Props
```
[the remark this answers is not in the document]
Comment (alanbsmith) [reply modules/preview-react/menu/stories/Menu.stories.mdx:125]: If you follow this format, you should be able to remove the rest of what's below.

File: modules/preview-react/menu/stories/examples/ManyItems.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import {Menu, MenuItem} from '@workday/canvas-kit-preview-react/menu';
+
+export const ManyItems = () => {
+  return (
+    <Menu title="Menu Titles">
+      {'One Two Three Four Five Six Seven Eight Nine Ten Eleven Twelve Thirteen Fourteen Fifteen Sixteen Seventeen'
```
Comment (alanbsmith) [inline modules/preview-react/menu/stories/examples/ManyItems.tsx:8]: Our [Menu design docs](https://design.workday.com/components/popups/menus) mention that menus should have 7-15 items. I'm not sure where the research came from on that. But it feels like we can align here and keep the example to 15. 

```suggestion
      {'One Two Three Four Five Six Seven Eight Nine Ten Eleven Twelve Thirteen Fourteen Fifteen'
```

**Review by alanbsmith — CHANGES_REQUESTED** [review]
Thanks for working on this, @sahlhoff! The example updates are a big improvement. Thanks for modernizing those. 😄 I made some copy and structure suggestions. Feel free to message me if any of it is unclear.

File: modules/preview-react/menu/stories/Menu.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+---
+
+#### `shouldClose: boolean`
+
+> Allows the onClose Menu callback to be fired after the menu item has been clicked
+
+Default: `true`
+
+---
```
Comment (alanbsmith) [inline modules/preview-react/menu/stories/Menu.stories.mdx:257]: We can remove these sections now.
```suggestion

```

File: modules/preview-react/menu/stories/Menu.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+- `tabindex={-1}`
+- `id`s following this pattern: `${MenuId}-${index}`
+
+#### Props
+
+<ArgsTable of={MenuItem} />
+
+## Specifications
+
+<Specifications file="Menu.spec.ts" name="Menu" />
```
[the remark this answers is not in the document]
Comment (sahlhoff) [reply modules/preview-react/menu/stories/Menu.stories.mdx:110]: Seems like we need to add it as an import...a little confusing being that `ArgsTable` and `ExampleCodeBlock` are globally available.

**Comment by sahlhoff** [comment]
@alanbsmith thank you for the review comments! should be good fo another look over if you get a chance :)

File: cypress/integration/Menu.spec.ts [hunk 6]
```diff
@@ -27,7 +27,7 @@ describe('Menu', () => {
 
   context('given "With Many Items" story is rendered', () => {
```
Comment (alanbsmith) [inline cypress/integration/Menu.spec.ts:28]:
```suggestion
  context('given the [Preview/Menu/React, Control Button] example is rendered', () => {
```

Comment (alanbsmith) [reply cypress/integration/Menu.spec.ts:28]: ^^ This will add a link in `Specifications` table to the story.

**Review by alanbsmith — APPROVED** [review]
Thanks for all the updates, @sahlhoff! 🥳

**Pull request merged**